syntax = "proto3";
import "google/protobuf/wrappers.proto";

package protos;

message TimeRange {
  uint64 time_geq = 1;
  uint64 time_lt = 2;
}

message Page {
  google.protobuf.UInt32Value limit = 1;
  google.protobuf.UInt32Value offset = 2;
}


message Meta {
  TimeRange time_range = 1;
  Page page = 2;
  google.protobuf.UInt32Value total_count = 3;
  google.protobuf.BoolValue show_inactive = 4;
}

message Message {
  string title = 1;
  string description = 2;
  string traceback = 3;
}

message ErrorMessage {
  google.protobuf.BoolValue error = 1;
  Message message = 2;
}

